app/src/main/java/com/monstarlab/features/login/ui/LoginViewModel.kt

```diff
@@ -2,26 +2,48 @@ package com.monstarlab.features.login.ui
 
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
-import com.monstarlab.core.extensions.LoadingAware
-import com.monstarlab.core.extensions.ViewErrorAware
+import com.monstarlab.core.error.toError
 import com.monstarlab.features.login.domain.usecase.LoginUseCase
 import dagger.hilt.android.lifecycle.HiltViewModel
-import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 @HiltViewModel
 class LoginViewModel @Inject constructor(
     private val loginUseCase: LoginUseCase
-) : ViewModel(), ViewErrorAware, LoadingAware {
+) : ViewModel() {
 
-    val loginResultFlow: MutableSharedFlow<Boolean> = MutableSharedFlow()
+    private val _stateFlow: MutableStateFlow<LoginState> = MutableStateFlow(LoginState())
 
-    fun login(email: String, password: String) {
+    val stateFlow: StateFlow<LoginState> get() = _stateFlow.asStateFlow()
```

Comment:
I believe that  `val stateFlow = _stateFlow.asStateFlow()` is enough, we don't need to instantiate new flow everytime it's accessed

Reply:
Addressed 👍